File: python/997. Find the Town Judge/solution.py
class Solution:
    def findJudge(self, n: int, trust: List[List[int]]) -> int:
        # outgoing and incoming

        scores = [0] * (n + 1)

        for connection in trust:
            scores[connection[0]] -= 1
            scores[connection[1]] += 1

        for i in range(1, len(scores)):
            if scores[i] == n - 1:
                return i

        return -1

File: java/997. Find the Town Judge/Solution.java
class Solution {
    public int findJudge(int n, int[][] trust) {
        // if there is a judge, the outgoing connection will be 0
        // and incoming connection will be n - 1
        int[] scores = new int[n + 1];

        for (int[] connection : trust) {
            scores[connection[0]]--;
            scores[connection[1]]++;
        }

        for(int i = 1; i < scores.length; i++) {
            if (scores[i] == n - 1) return i;
        }
        return -1;
    }
}